### ISSUE TITLE
Infinite recursion of Pyreverse when numpy.array is in class

### ISSUE THREAD

**Issue by SpookyYomo** [pylint-dev/pylint#9681]
### Bug description
MWE:
```python
import numpy as np

class A:
    foo = np.array([1, 2])
```
and then run the following:

```
pyreverse -ASmy fail.py
```

### Command used

```shell
pyreverse -ASmy fail.py
```

### Pylint output

After about 3 minutes of waiting...
<details>
<summary>Ouput:
</summary>

```python
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c06773a6600>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c06773c33e0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0677287e00>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c067729c8c0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0677109b20>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0677095790>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c06770ae540>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0677072f60>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676fe39e0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676e58440>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676d1fc80>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676d68ad0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676d355b0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676d5dfd0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676d429f0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676bfa2d0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676bff080>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676ae7aa0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676af0560>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c0676a84fe0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c06768a0800>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c06768b5670>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c06768aa030>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c06768d2a50>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c067667b410>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c06763a9ca0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c06763c66c0>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/transforms.py:118: UserWarning: Astroid was unable to transform Return(value=<Tuple.tuple l.31 at 0x7c06763d70e0>).
Some functionality will be missing unless the system recursion limit is lifted.
...
  File "/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/rebuilder.py", line 1154, in _visit_functiondef
    position=self._get_position_info(node, newnode),
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/my_username/.local/share/pyenv/versions/3.12.1/lib/python3.12/site-packages/astroid/rebuilder.py", line 153, in _get_position_info
    return Position(
           ^^^^^^^^^
  File "<string>", line 1, in <lambda>
KeyboardInterrupt
```

</details>

### Expected behavior

The pyreverse module should generate a UML class `A` with `foo`. However, it loops forever when processing `np.array`. I have been unable to create a MWE without using `np.array`.

### Pylint version

```shell
pylint 3.2.2
astroid 3.2.2
Python 3.12.1 (main, Feb  9 2024, 14:00:11) [GCC 13.2.1 20230801]

numpy 1.26.4
```

### OS / Environment

OS: Arch Linux x86_64
Kernel: Linux 6.9.2-arch1-1

### Additional dependencies

```python
astroid==3.2.2
dill==0.3.8
isort==5.13.2
mccabe==0.7.0
numpy==1.26.4
platformdirs==4.2.2
pylint==3.2.2
tomlkit==0.12.5
```

**Comment by SpookyYomo** [comment]
Running `pyreverse -s 3 -my fail.py` followed by `dot -Tpng classes.dot` yields the following image
![ndarray](https://github.com/pylint-dev/pylint/assets/48710653/73565f40-b9d5-4cf1-bd40-fdd807021b1e)


ndarray does appear to be deeply recursive and pyreverse is unable to break out.

**Comment by chillerb** [comment]
Sorry, I just can confirm that this is still an issue, that I actually just encountered in my project:

```sh
$ pyreverse --version
pyreverse is included in pylint:
pylint 3.3.7
astroid 3.3.10
Python 3.12.11 | packaged by conda-forge | (main, Jun  4 2025, 14:45:31) [GCC 13.3.0]
```

The recursion seems to stem from the array's `shape` property, if I am not mistaken.

<img width="363" height="635" alt="Image" src="https://github.com/user-attachments/assets/36e12295-fd0e-4b03-bd78-03ec83aa132c" />

**Comment by Pierre-Sassoulas** [comment]
Look like a duplicate of https://github.com/pylint-dev/pylint/issues/3602
